assets/src/edit-story/components/inspector/prepublish/prepublishChecklistProvider.js

```diff
@@ -47,46 +47,87 @@ function PrepublishChecklistProvider({ children }) {
     return { ...story, pages };
   });
 
+  const isStoryLoaded = story?.pages?.length > 0;
+  const isChecklistEmpty = checkpointState === PPC_CHECKPOINT_STATE.NO_ISSUES;
+
   const [currentList, setCurrentList] = useState([]);
+  const [isChecklistReviewRequested, setIsChecklistReviewRequested] = useState(
+    false
+  );
 
   const handleRefreshList = useCallback(async () => {
     const pagesWithSize = story.pages.map((page) => ({
       ...page,
       pageSize,
     }));
 
-    // discern PPC types of messages based on checkpointState
-    let types;
-    switch (checkpointState) {
-      case PPC_CHECKPOINT_STATE.UNAVAILABLE:
-        types = [];
-        break;
-      default:
-        types = [
-          PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
-          PRE_PUBLISH_MESSAGE_TYPES.ERROR,
-          PRE_PUBLISH_MESSAGE_TYPES.WARNING,
-        ];
-        break;
-    }
-
     setCurrentList(
-      await getPrepublishErrors({ ...story, pages: pagesWithSize }, { types })
+      await getPrepublishErrors({ ...story, pages: pagesWithSize })
     );
-  }, [story, pageSize, checkpointState]);
+  }, [story, pageSize]);
 
   const prevPages = usePrevious(story.pages);
   const prevPageSize = usePrevious(pageSize);
 
   const refreshOnInitialLoad = prevPages?.length === 0 && story.pages?.length;
   const refreshOnPageSizeChange = prevPageSize?.width !== pageSize?.width;
 
+  const highPriorityLength = useMemo(
+    () =>
+      isStoryLoaded &&
+      currentList.filter(
+        (current) => current.type === PRE_PUBLISH_MESSAGE_TYPES.ERROR
+      ).length,
+    [currentList, isStoryLoaded]
+  );
+
+  const recommendedLength = useMemo(
+    () =>
+      isStoryLoaded &&
+      currentList.filter((current) =>
+        current.type.indexOf([
+          PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
+          PRE_PUBLISH_MESSAGE_TYPES.WARNING,
+        ])
+      ).length,
+    [currentList, isStoryLoaded]
+  );
```

Comment:
should we set these to `0` instead of `false` if the story hasn't been loaded yet?

Reply:
Actually the false is important because it shows the checklist hasn't run at all yet. Later on (https://github.com/google/web-stories-wp/pull/7298/files#diff-540f459f1e6797cd816d4fd230fad7f5fcd4934082d3dcf71d67632bf01a5675R115) we're checking for `0` specifically to dispatch events.